bindings/java/src/main/java/org/openucx/jucx/ucp/UcpListener.java

```diff
@@ -31,6 +32,15 @@ public UcpListener(UcpWorker worker, UcpListenerParams params) {
         setNativeId(createUcpListener(params, worker.getNativeId()));
     }
 
+    /**
+     * Reject the incoming connection request and release associated resources. If
+     * the remote initiator endpoint has set an {@link UcpEndpointParams#setErrorHandler},
+     * it will be invoked with status {@link UcsConstants.STATUS#UCS_ERR_REJECTED}.
+     */
+    public void reject(UcpConnectionRequest connRequest) {
```

Comment:
maybe it should be UcpConnectionRequest.reject()?

Reply:
We need to know a listener pointer. One option is to add:
```
ucs_status_t ucp_conn_request_reject(ucp_conn_request_h conn_request) 
{
   ucp_listener_reject(conn_request->listener, conn_request);
}
```

to UCP API or to use ` ucp_listener_reject(conn_request->listener, conn_request);` in jucx API. WDYT?

Reply:
prefer to not change UCP API, and also not access internal pointer of conn_request from JNI
how about storing listener object in UcpConnectoinRequest?

Reply:
There's no way to get a listener from conn_handler callback (except of again using conn_request->listner).

Reply:
Or i can add in #7237 an option to query listener from con_request - would be compromise - not adding new API, using ucp_conn_request_query(...

Reply:
can you pass the UcpListener object in https://github.com/openucx/ucx/blob/f33df4ad5fa24b8f2e5ee2e01648bbf2fb9ccbb7/bindings/java/src/main/native/listener.cc#L47 instead of UcpListenerConnectionHandler?
then fetch connhandler from listener in jucx_connection_handler

Reply:
`jucx_conn_handler` is a generic callback that can be used for **several** listeners

Reply:
Also, it's passed as a listener params before the listener is created

Reply:
but we create UCP listener from UcpListener constructor
https://github.com/openucx/ucx/blob/f33df4ad5fa24b8f2e5ee2e01648bbf2fb9ccbb7/bindings/java/src/main/java/org/openucx/jucx/ucp/UcpListener.java#L31
don't we already have object then?

Reply:
Ok, i got it